### PR TITLE
Mopping with just a bucket

### PR DESCRIPTION
**Pull request by kingofkosmos**
Enables wetting the mop from a normal bucket.

Also wetting the mop in the sink.

It felt a bit weird that the mop wouldn't work with a normal bucket (so this just because realism and usability). Trying to think how this could affect gameplay; this could be seen as a nerf to the janicart since now it's possible to have all your mopping needs in your backpack and not have to drag around a big thing that can be easily stolen.

### CODE REVIEW

**Comment by phil235** [comment]
Try to put both attackby procs together please, we only need one.

**Comment by WJohn** [comment]
I feel like we should be able to wet the mop from a lot of other sources, honestly. Why not be able to use a sink, beaker, or glass with water in it?

**Comment by MrStonedOne** [comment]
Using a mob on a wet floor should wet it and un wet the floor

**Comment by Miauw62** [comment]
mso what would be the point
if you're not leaving wet floors everywhere for people to slip over what's the point of playing janitor in the first place. just put up some signs and point at them when people slip

**Comment by kingofkosmos** [comment]
Getting water from sinks could be cool, I'll look into that. I'm not too sure about beakers and drinking glasses though, feels like they would be too small for the mop to fit in.

**Comment by francinum** [comment]
@kingofkosmos How about large beakers?

**Comment by phil235** [comment]
Wetting a mop with a beaker would be weird.

**Comment by kingofkosmos** [comment]
Added wetting in the sink, you now get the message "You wet the mop in the sink." and "You start washing the mop..." so it's clear that if you want a bloody mop washed, you'll have to stay still, but wetting is instant.

I had an idea if to wet a mop when it gets under a turned on shower, but that's probly too advanced stuff for me.

**Comment by phil235** [comment]
There is still two /obj/item/weapon/reagent_containers/glass/bucket/attackby

**Comment by kingofkosmos** [comment]
Fixed.

**Comment by Dorsisdwarf** [comment]
Is this going to be merged at some point?

File: code/modules/reagents/reagent_containers/glass.dm [hunk 1]
```diff
@@ -234,10 +234,20 @@
...
+		qdel(O)
+	if(istype(O, /obj/item/weapon/mop))
+		if(reagents.total_volume < 1)
+			user << "<span class='warning'>[src] is out of water!</span>"
+		else
+			reagents.trans_to(O, 5)
+			user << "<span class='notice'>You wet [O] in [src].</span>"
+			playsound(loc, 'sound/effects/slosh.ogg', 25, 1)
+	else if(isprox(O))
+		user << "<span class='notice'>You add [O] to [src].</span>"
```
Comment (phil235) [inline code/modules/reagents/reagent_containers/glass.dm]: there's two isprox() checks, on line 238 and 248

**Comment by kingofkosmos** [comment]
There! Should be ok now.

**Pull request merged**